docs/content/guides/dagster/pythonic-resources.mdx

```diff
@@ -0,0 +1,337 @@
+- **Share implementations across multiple ops or assets** - When multiple ops access the same external services, resources provide a standard way to structure your code to share the implementations.
+
+---
+
+## Using resources
+
+### Getting started using resources
+
+A resource can be any Python object which you would like to share between software-defined assets or ops. Assets and ops specify resource dependencies by annotating the resource as a parameter to the asset or op function. The parameter type should be wrapped in a <PyObject object="Resource"/> annotation to indicate that it is a resource and not an upstream asset or op input.
+
+To provide resource values to your assets and ops, attach them to your <PyObject object="Definitions" /> call. These resources are automatically passed to the function at runtime.
+
+<TabGroup persistentKey="assetsorops">
+<TabItem name="Using software-defined-assets">
+
+#### Using software-defined assets
+
+Here, we define a simple string resource representing a URL which we use in a software-defined asset. We specify a value for the URL in the <PyObject object="Definitions" /> call.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_new_resources_assets_defs endbefore=end_new_resources_assets_defs
+from dagster import asset, Definitions
+from dagster._core.definitions.resource_annotation import Resource
+
+
+@asset
+def data_from_url(data_url: Resource[str]) -> Dict[str, Any]:
+    return requests.get(data_url).json()
+
+
+defs = Definitions(
+    assets=[data_from_url],
+    resources={"data_url": "https://dagster.io"},
+)
+```
+
+</TabItem>
+<TabItem name="Using ops and jobs">
+
+#### Using ops
+
+Here, we define a simple string resource representing a URL which we use in an op. We specify a value for the URL in the <PyObject object="Definitions" /> call.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_new_resources_ops_defs endbefore=end_new_resources_ops_defs
+from dagster import op, Definitions, job
+from dagster._core.definitions.resource_annotation import Resource
+
+
+@op
+def print_data_from_resource(data_url: Resource[str]):
+    print(requests.get(data_url).json())
+
+
+@job
+def print_data_from_url_job():
+    print_data_from_resource()
+
+
+defs = Definitions(
+    jobs=[print_data_from_url_job],
+    resources={"data_url": "https://dagster.io"},
+)
+```
+
+</TabItem>
+</TabGroup>
+
+### Configurable resources
+
+In many cases, resources need to be [configurable](/concepts/configuration/config-schema). For example, a resource representing a connection to an external service may need a configurable username and password. Configurable resources can be defined by extending the <PyObject object="ConfigurableResource"/> class. The attributes on the class are used to define the configuration schema, which lets you customize behavior through run configuration.
```

Comment:
We should answer the question, why "configuration" rather than just plain ole python parameter passing here